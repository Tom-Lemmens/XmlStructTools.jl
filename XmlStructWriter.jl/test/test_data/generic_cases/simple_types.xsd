<?xml version="1.0"?>
<schema xmlns="http://www.w3.org/2001/XMLSchema" xmlns:TestSimpleTyping="TestSimpleTyping" targetNamespace="TestSimpleTyping">

    <element name="document" type="TestSimpleTyping:documentType"/>

    <simpleType name="TestSimpleType1">
        <annotation>
            <documentation>An example of a simple xsd type.</documentation>
        </annotation>
        <restriction base="string">
            <maxLength value="4"/>
            <pattern value="([0-9A-Z]{4})?"/>
        </restriction>
    </simpleType>

    <simpleType name="TestSimpleType2">
        <annotation>
            <documentation>An example of a simple xsd type.</documentation>
        </annotation>
        <restriction base="integer">
            <minInclusive value="100"/>
            <maxInclusive value="500"/>
        </restriction>
    </simpleType>

    <simpleType name="TestSimpleType3">
        <annotation>
            <documentation>An example of a redundant simple xsd type.</documentation>
        </annotation>
        <restriction base="boolean"/>
    </simpleType>

    <simpleType name="TestSimpleType4">
        <annotation>
            <documentation>An example of a simple xsd type.</documentation>
        </annotation>
        <restriction base="positiveInteger">
            <minInclusive value="10"/>
            <maxInclusive value="50"/>
        </restriction>
    </simpleType>

    <simpleType name="TestSimpleType5">
        <annotation>
            <documentation>An example of a simple xsd type.</documentation>
        </annotation>
        <restriction base="decimal">
            <minInclusive value="-200"/>
            <maxInclusive value="200"/>
        </restriction>
    </simpleType>


    <complexType name="TestComplexType1">
        <sequence>
            <element name="TestElement1" type="TestSimpleTyping:TestSimpleType1"/>
            <element name="TestElement2" type="TestSimpleTyping:TestSimpleType2"/>
            <element name="TestElement3" type="TestSimpleTyping:TestSimpleType3"/>
            <element name="TestElement4" type="TestSimpleTyping:TestSimpleType4"/>
            <element name="TestElement5" type="TestSimpleTyping:TestSimpleType5"/>
        </sequence>
    </complexType>

    <complexType name="TestComplexType2">
        <annotation>
            <documentation>An example of a type with string based elements both with and without default values.</documentation>
        </annotation>
        <sequence>
            <element name="TestElement11" type="TestSimpleTyping:TestSimpleType1" default="DEFA"/>
            <element name="TestElement12" type="TestSimpleTyping:TestSimpleType1"/>
            <element name="TestElement21" type="TestSimpleTyping:TestSimpleType2" default="100"/>
            <element name="TestElement22" type="TestSimpleTyping:TestSimpleType2"/>
        </sequence>
    </complexType>


    <complexType name="documentType">
        <sequence>
            <element name="TestElement1" type="TestSimpleTyping:TestComplexType1"/>
            <element name="TestElement2" type="TestSimpleTyping:TestComplexType2"/>
        </sequence>
    </complexType>

</schema>