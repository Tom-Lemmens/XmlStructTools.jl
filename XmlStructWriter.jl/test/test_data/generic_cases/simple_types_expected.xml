<?xml version="1.0" encoding="utf-8"?>

<TestSimpleTyping:document xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="TestSimpleTyping simple_types.xsd" xmlns:TestSimpleTyping="TestSimpleTyping">
    <TestElement1>
        <TestElement1>AAAA</TestElement1>
        <TestElement2>125</TestElement2>
        <TestElement3>true</TestElement3>
        <TestElement4>25</TestElement4>
        <TestElement5>5.55</TestElement5>
    </TestElement1>
    <TestElement2>
        <TestElement1>DEFA</TestElement1>
        <TestElement2></TestElement2>
    </TestElement2>
</TestSimpleTyping:document>